code/modules/mob/living/living.dm

```diff
@@ -746,16 +746,17 @@
 /mob/living/proc/float(on)
 	if(throwing)
 		return
-	var/fixed = 0
+	var/fixed = FALSE
 	if(anchored || (buckled && buckled.anchored))
-		fixed = 1
+		fixed = TRUE
 	if(on && !floating && !fixed)
 		animate(src, pixel_y = pixel_y + 2, time = 10, loop = -1)
-		floating = 1
+		sleep(10)
```

Comment:
Tested this; not really a problem because of how floating conditionally handles things.

Going into space with magboots on means you don't float--having the magboots off means you do float.

Likewise, if you're a flying mob and toggle magboots on or off you'll still float---in space or not.